### ISSUE TITLE
Replace submodules of `rhea` and `magnet` with a `pip` VCS (`git+`) method in the env.yml

### ISSUE THREAD

**Issue by ppreshant** [treangenlab/somatem#118]
# Goal 
- Need to make the install more robust by replacing the submodule based of rhea and magnet with a `pip git+` dependancy based install 
- Noticed that the submodule directory is not carried over when setting up the package from bioconda. So will need this as a part of #109

Example yml
```yaml
name: magnet
channels:
    - conda-forge
    - bioconda
dependencies:
    - python
    - ncbi-datasets-cli 
    - minimap2
    - samtools
    - biopython
    - pandas
    - ete3
    - git
    - bwa
    - scikit-learn
    - numpy
    - fastani
    - pip
    - pip:
      - legacy-cgi
      - git+https://github.com/treangenlab/magnet.git
```
- Can include a version tag or peg it to main branch like this 
```
git+https://github.com/org/nf-modules.git@v1.0.0
git+https://github.com/org/custom-modules.git@main
```

## Slack note
Ok, read up a bit more about the [pip VCS](https://pip.pypa.io/en/stable/topics/vcs-support/) (git +) method you suggested. Unfortunately, it will not work for our use case unless it is a `python package`.  Also Nextflow might have issues cacheing the pip installed repo so it may not be the best idea. [deepseek analysis](https://chat.deepseek.com/share/2wbimno76qrd2e8r6v)

• Magnet has a setup.py so it would work here
• rhea doesn't seem to be a package but just standalone python script. 
If you want to make a setup.py for rhea, and add tags of `v1.0` in both repos, then I can try this method sometime this week.

>  When you do:
> bash
> ```pip install git+<https://github.com/user/repo.git>```
> You're installing a *Python package*, which means:
> The repo must have one of these: `setup.py` / `pyproject.toml` / `setup.cfg`
> 1. It gets **installed to**: `{conda_env}/lib/pythonX.Y/site-packages/`
> 2. Python packages get "flattened" - not preserving original repo structure
> *If it's Just a Script Repository (No setup.py):*
> This won't work with `pip git+`!

Can locate the package after pip install using these 
```py
import sys
import pkg_resources

# Method 1: Using pkg_resources
package_path = pkg_resources.resource_filename('mypackage', '')
```

## Pre-setup
Can create a proper python module with this structure
```
mytool/
├── setup.py
├── pyproject.toml
├── mytool/
│   ├── __init__.py
│   └── something.py  # Your script as a module
└── README.md
```

[Slack Message](https://treangenlab.slack.com/archives/D08HP4K72QJ/p1770073385423539?thread_ts=1770065252.497149&cid=D08HP4K72QJ)

## Plan 
- [ ] test the pip method on `magnet` which has the `setup.py` added by Austin recently
  - [ ] (_if necessary_) Add `manifest.toml` in `magnet` 
- [ ] Add `setup.py` and `manifest.toml` to `rhea`  repo

**Comment by ppreshant** [comment]
## Notes 
Using branch `submod_to_pip_i118` based on `bioconda_cleanup_i109`  
- chosen base other than main so I can test it with the bioconda as well)
 
- testing magnet: encountered error with the older version I imposed `scikit-learn=0.23.2` since the `affinity`  was replaced by `metric` as the latest version (1.7 or [1.8](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.AgglomerativeClustering.html#sklearn.cluster.AgglomerativeClustering)) requires by Austin's [commit](https://github.com/treangenlab/magnet/commit/3839211c8e64cc5aae96fe1517aefa4b8e9888af#diff-25ec87bb90527db6f20dd7472b725dcfb29ab5fe404ad84d5b94bddf6f587b9f) in magnet.py
```py
model = AgglomerativeClustering(affinity='precomputed', n_clusters=None, compute_full_tree=True,
    model = AgglomerativeClustering(metric='precomputed', n_clusters=None, compute_full_tree=True,    model = AgglomerativeClustering(metric='precomputed', n_clusters=None, compute_full_tree=True,
```
  - Error dir: `somatem/work/4f/221d0f811fcbdfaf2d56096987e2771` in env `/home/Users/pbk1/micromamba/nextflow-envs/env-d49602fbfb936ba6f89f6a62cc96519b`

- `rhea` test logs:
```groovy
~/somatem$ nextflow run test-modules/rhea_test.nf
Nextflow 26.04.0 is available - Please consider updating your version to it

 N E X T F L O W   ~  version 25.10.4

Launching `test-modules/rhea_test.nf` [special_angela] DSL2 - revision: 7febadb80b

[-        ] RHEA -
executor >  local (1)                                                                                                                    executor >  local (1)
[c5/0506b2] RHEA (t0_t1) [100%] 1 of 1 ✔
Completed at: 08-May-2026 21:26:59
Duration    : 2m 45s
CPU hours   : 0.5
Succeeded   : 1
```
